packages/liveblocks-core/src/comments/comment-body.ts

```diff
@@ -632,3 +632,112 @@ export async function stringifyCommentBody(
 
   return blocks.join(separator);
 }
+
+export type TransformCommentBodyOptions<U extends BaseUserMeta = DU> = {
+  /**
+   * Which format to transform the comment body to.
+   */
+  format?: "html" | "json";
+  /**
+   * A function that returns info from user IDs.
+   */
+  resolveUsers?: (
+    args: ResolveUsersArgs
+  ) => OptionalPromise<(U["info"] | undefined)[] | undefined>;
+};
+
+export type CommentBodyJsonMention = {
+  type: "mention";
+  user: string;
+};
+export type CommentBodyJsonLink = CommentBodyLink;
+export type CommentBodyJsonText = CommentBodyText;
+
+export type CommentBodyJsonInlineElement =
+  | CommentBodyJsonMention
+  | CommentBodyJsonLink
+  | CommentBodyJsonText;
+export type CommentBodyJsonParagraph = {
+  type: "paragraph";
+  children: CommentBodyJsonInlineElement[];
+};
+export type CommentBodyJsonBlockElement = CommentBodyJsonParagraph;
+/**
+ * A `CommentBodyJson` represents the content of a `CommentBody`
+ * with resolved users or not and with absolute urls
+ */
+export type CommentBodyJson = {
+  version: 1;
+  content: CommentBodyJsonBlockElement[];
+};
+
+/**
+ * Transform a `CommentBody` into either an HTML format (HTML safe string)
+ * or into a JSON format and resolves users.
+ *
+ * By default the format is set to "HTML".
+ */
+export async function transformCommentBody(
+  body: CommentBody,
+  options?: TransformCommentBodyOptions<BaseUserMeta>
+): Promise<string | CommentBodyJson> {
+  const format = options?.format ?? "html";
+
+  switch (format) {
+    case "html":
+      return await stringifyCommentBody(body, {
+        format: "html",
+        resolveUsers: options?.resolveUsers,
+      });
+    case "json": {
+      const resolvedUsers = await resolveUsersInCommentBody(
+        body,
+        options?.resolveUsers
+      );
+
+      const content: CommentBodyJsonBlockElement[] = body.content.map(
+        (block) => {
+          return {
+            type: "paragraph",
+            children: block.children.map(
+              (inline): CommentBodyJsonInlineElement => {
+                if (isCommentBodyMention(inline)) {
+                  const user = resolvedUsers.get(inline.id);
+                  return {
+                    type: "mention",
+                    user: `@${user ? user.name : inline.id}`,
+                  };
+                } else if (isCommentBodyLink(inline)) {
+                  return {
+                    type: "link",
+                    url: toAbsoluteUrl(inline.url) ?? inline.url,
+                    text: inline.text,
+                  };
+                } else if (isCommentBodyText(inline)) {
+                  return { ...inline };
+                }
+
+                // NOTE: safeguard in case we support other types than "text", "link", or "mention"
+                throw new TransformCommentBodyError(
+                  `unsupported comment body inline element: ${JSON.stringify(inline)}`
+                );
```

Comment:
From my experience failing silently is not a very good DX. 
I may be too cautious here but I think it's worth to throw an error in case we miss to update the code here. 

From a developer standpoint it will be more valuable to us as we do not have any error reporters plugged and as developers will know what happens. And it may make them want to contribute to Liveblocks or report an error. 

It won't create frustration using us as they will know there is an error and why their code fails.

Reply:
This is one of the rare cases where we should not throw an error if we encounter an unknown node. It doesn't need to be silent, though. console.warn seems appropriate (explaining they need to upgrade or there is a bug in our system).

If we decide to add a new node to our schema on the backend, it will crash our customers' webhooks. Not rendering a node is better than crashing our customers' notification system.

Reply:
Arf I see what you mean 👍🏻 

So we're assuming that developers may not write a webhook with a `try/catch` clause. 
From this point of view it seems faire 👍🏻 Even if we take developers a bit too much by the hand from my point of view 😅